src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -65,13 +65,14 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             try
             {
                 await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
-                await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
                 this.RegisterService<ICSharpTempPECompilerService>(async ct =>
                 {
                     var workspace = this.ComponentModel.GetService<VisualStudioWorkspace>();
+                    var metadataService = workspace.Services.GetService<IMetadataService>();
+
                     await JoinableTaskFactory.SwitchToMainThreadAsync(ct);
-                    return new TempPECompilerService(workspace.Services.GetService<IMetadataService>());
+                    return new TempPECompilerService(metadataService);
```

Comment:
>TempPECompilerService [](http://example.com/codeflow?start=31&length=21)

Does this ctor really need to be called on the main thread?

Reply:
(Same question applies for the VisualBasicPackage.InitializeAsync)

Reply:
Tagging @davkean just to make sure this question has been seen

Reply:
No it does not.